tests/phpunit/tests/fonts/font-face/wpFontFaceResolver/getFontsFromThemeJson.php

```diff
@@ -37,20 +37,59 @@ public function test_should_return_all_fonts_from_theme() {
 		$this->assertSame( $expected, $actual );
 	}
 
+	public function test_should_return_all_fonts_from_all_theme_origins() {
+		switch_theme( static::FONTS_THEME );
+
+		// Add a filter to add fonts to the theme.json custom origin.
+		add_filter(
+			'wp_theme_json_data_theme',
+			function ( $theme_json_data ) {
```

Comment:
Optimization 
```suggestion
			static function ( $theme_json_data ) {
```

Reply:
Hmm wait ... this is hooking into a filter. But I don't think the filter gets automatically cleared of its callbacks at teardown. 

So to address this, need to:

1. Move the callback up and assign it to a variable.
```
$add_fonts_callback = static function ( ...
```
3. Hook the variable into the `add_filter()`.
4. And before doing an assertion, unhook the callback
```
remove_filter( 'wp_theme_json_data_theme', $add_fonts_callback );
```

This will avoid cross-polluting other tests that run after this one.

Reply:
Removed here: https://github.com/WordPress/wordpress-develop/pull/6161/commits/25e6492d50c03d62636b9d982cb7197be7c7f71d